### ISSUE TITLE
Importing an icalendar file with weekly repeated events scheduled for weekdays, results in all the events being scheduled on Sundays

### ISSUE THREAD

**Issue by gphilip** [meeting-room-booking-system/mrbs-code#3485]
**Describe the bug**
If I import an icalendar (.ics) file that has repeating events specified using the "RRULE" element, where 

- the events are scheduled to occur on specific weekdays using the BYDAY directive
- these events repeat every week till some ENDDATE
 
 then in the resulting MRBS schedule all the events are shown as being scheduled on Sundays.  

**To Reproduce**
1. Set up an MRBS instance with the proper areas, rooms, and event types.
2. Import a .ics icalendar file which has RRULE entries that describe weekly repeating events which are scheduled on specific weekdays (MO to FR) using the BYDAY directive of RRULE. Example:

 
> BEGIN:VEVENT
> SUMMARY:BugDemo
> DTSTART;TZID=Asia/Kolkata:20230810T120000
> DTEND;TZID=Asia/Kolkata:20230810T120100
> RRULE:FREQ=WEEKLY;UNTIL=20231126T235900;BYDAY=TH
> DESCRIPTION:Demonstrating ICS Import Bug
> LOCATION:ICS/BugReport
> X-MRBS-TYPE:Bug Report
> END:VEVENT

3. Look at the resulting MRBS schedule
4. All the events are shown as being scheduled on Sundays, irrespective of the day specified by the BYDAY directive.

**Expected behavior**
The repeating events should be scheduled on the respective days as specified by the BYDAY directives.

**Browser details (please complete the following information):**
 - OS: Linux
 - Browser: Firefox
 - Version: 115.0.3

**Server details (please complete the following information):**
 - MRBS version 1.11.3
 - Web server:  Apache 2.4.54
 - OS: Linux
 - PHP version: 7.4.30
 - Database used: MySQL 10.5.15-MariaDB

**Possible reason for this bug**
Starting at [line 397 of web/functions_ical.inc](https://github.com/meeting-room-booking-system/mrbs-code/blob/main/web/functions_ical.inc#L399) there is the following code:

```
case 'WEEKLY':
      $repeat_rule->setType(RepeatRule::WEEKLY);
      if (isset($rules['BYDAY']))
      {
        $byday_days = explode(',', $rules['BYDAY']);
      }
      else
      ... // Stuff not relevant for this bug
      ...
      $repeat_rule->setDays($byday_days);
```

As per [the icalendar specification](https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.10), valid values for 'BYDAY' look like 'SU', 'MO', and so on. Assuming that the repeat rule has just one 'BYDAY' value, say 'TH', the variable $byday_days will end up being an array with the one string 'TH'.

The [code for RepeatRule->setDays()](https://github.com/meeting-room-booking-system/mrbs-code/blob/main/web/lib/MRBS/RepeatRule.php#L163) does the following:

```
public function setDays(array $days) : void
  {
    foreach ($days as $day)
    {
      // Cast to int
      $this->days[] = (int) $day;
      // Check it's a valid day
      if (($day < 0) || ($day > 6))
      {
        throw new InvalidArgumentException("Invalid day of the week '$day'");
      }
    }
    // We need the repeat days to be in order
    sort($this->days);
  }
```

The casting from the string 'TH' to int results in the integer 0, and so this recurring event gets scheduled for Sundays.

I don't know enough PHP to offer a reliable fix, sorry about that.

**Comment by campbell-m** [comment]
Thanks for spotting this.  The bug was introduced since 1.11.3 in a recent merge and I have now fixed it in [b9d72dd](https://github.com/meeting-room-booking-system/mrbs-code/commit/b9d72ddc77c84abdd9cfe4f61a104cfc98bed7f0).  There was also another bug when BYDAY was not specified which is now fixed in [5518689](https://github.com/meeting-room-booking-system/mrbs-code/commit/5518689cad699fef924d2eb49e9869386d702cdd).

**Comment by gphilip** [comment]
Thank you for fixing this. I currently have no means of checking this; I used the workaround of replacing "MO" with "1" etc. in the .ics source, to make the import work. 

I don't know how bug closing works here. If you feel that the fix closes the bug, please feel free to close it.
